### PR TITLE
[24.2] Workflow Inputs Activity

### PR DESCRIPTION
**Pull request by ElectronicBlueberry**
Moves inputs to it's own activity.
This will likely break a bunch of tests. Here's a picture:

image

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by jdavcs** [comment]
This needs to target the dev branch, right? Also, the milestone should be 25.0?

**Comment by dannon** [comment]
Yes, @davelopez has the right recap, we want this in 24.2.

**Comment by jmchilton** [comment]
"Single Dataset Input" seems like the wrong language for the top item description given that it can be a multiple="true" input. Feels cosmetic though - the backend changes all make sense to me and it is very good looking though. I like it and I appreciate the quick turn around for getting it into the release. Thank you so much!

File: lib/galaxy/webapps/galaxy/api/workflows.py [hunk 1]
```diff
@@ -531,12 +531,12 @@ def build_module(self, trans: GalaxyWebTransaction, payload=None):
...
         trans.workflow_building_mode = workflow_building_modes.ENABLED
         module = module_factory.from_dict(trans, payload, from_tool_form=True)
-        if "tool_state" not in payload:
-            module_state: Dict[str, Any] = {}
-            errors: ParameterValidationErrorsT = {}
-            populate_state(trans, module.get_inputs(), inputs, module_state, errors=errors, check=True)
-            module.recover_state(module_state, from_tool_form=True)
-            module.check_and_update_state()
+
+        module_state: Dict[str, Any] = {}
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/api/workflows.py:535]: Why this change ? I think this might break cloning with a passed tool state ? See https://github.com/galaxyproject/galaxy/pull/6085/files

Comment (mvdbeek) [reply lib/galaxy/webapps/galaxy/api/workflows.py:535]: It all seems to work fine in testing, and I see this fixes accessing module_state if the `tool_state` is in fact passed. I assume this hasn't caused problems because we stopped passing `tool_state` back to the build_module endpoint (all changes come through `payload.get("inputs", {})` ?) and now we only do this for inputs, so I think that's fine ? On the next occasion we should expand tests for this endpoint.

File: lib/galaxy/webapps/galaxy/controllers/workflow.py [hunk 2]
```diff
@@ -168,35 +161,6 @@ def editor(self, trans, id=None, workflow_id=None, version=None, **kwargs):
             else:
                 new_workflow = True
 
-        # The following query loads all user-owned workflows,
```
Comment (mvdbeek) [inline lib/galaxy/webapps/galaxy/controllers/workflow.py:171]: 🎉 this is so much snappier on my test instance with thousands of workflows ...
